.github/workflows/trivy.yml

```diff
@@ -9,12 +9,25 @@ on:
     - cron: '45 8 * * *'
   push:
     branches: ["master", "release/**"]
+    paths:
+      - '**/go.mod'
+      - '**/pom.xml'
+      - '**/*gradle.lockfile'
+      - '**/*.sbt.lock'
+      - '**/requirements.txt'
+      - '**/poetry.lock'
+      - '**/Pipfile.lock'
+      - '**/*trivy*'
   pull_request:
     branches: ["master", "release/**"]
     paths:
       - '**/go.mod'
       - '**/pom.xml'
+      - '**/*gradle.lockfile'
+      - '**/*.sbt.lock'
       - '**/requirements.txt'
+      - '**/poetry.lock'
+      - '**/Pipfile.lock'
       - '**/*trivy*'
```

Comment:
Is Trivy expensive to run?  If not, why do we bother with these path filters and not just run it on every PR and landing so that new issues are caught as soon as possible?

Who is going to know to update these path filters if/when Trivy expands their coverage to other languages, or even when some already supported language changes the file that Trivy should use, etc.?

I suppose the scheduled run will catch these kinds of changes, but again, without a process/responsible person put in place to monitor the Security Tab, these changes will go unnoticed and we will not be performing complete Trivy scans on PRs or even on landings.

Reply:
I have observed some limits on the Trivy database access.
(https://github.com/daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21)

Everything that breakes the PR build, but is not related to PR-related changes is VERY expensive from the PR's author perspective. Moreover, the same problem may hit several PRs in parallel. Who will be responsible to fix it?

Trivy is configured intentionally to detect CVEs in Python/Go/Java. This defines scope of files to be scaned. I do not expect that this will change soon. 

Why we can not treat `schedule` in the same way as we treat daily tests?
I expect that one scan per day is enough in normal case.

Reply:
> I have observed some limits on the Trivy database access. ([daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21](https://github.com/daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21))

44000/minute seems like a lot.  Are we sure something did not run amok?
 
> Everything that breakes the PR build, but is not related to PR-related changes is VERY expensive from the PR's author perspective.

No more than the status quo.  Just look at the **isort** action/check that has been broken for a few days last week.  It finally (and only, I suspect!) got updated because somebody noticed it in their PR and started asking about it/getting it fixed.  If it did not break a PR I am pretty sure it would still be broken and nobody noticing.

> Moreover, the same problem may hit several PRs in parallel. Who will be responsible to fix it?

Just like the **isort** problem last week.  Somebody will raise their PR being broken because of it and somebody jumps in to fix it.

> Trivy is configured intentionally to detect CVEs in Python/Go/Java. This defines scope of files to be scaned. I do not expect that this will change soon.

Which is even worse.  Because when a long time goes by and they do finally change something, nobody remembers that we need to change on our end also to capture that new something.
 
> Why we can not treat `schedule` in the same way as we treat daily tests? I expect that one scan per day is enough in normal case.

Phil has been assigned ownership of monitoring daily scans.  Who will be assigned daily monitoring of the Security Tab?  This is something that needs to be addressed before we introduce a security scan that may **only** publish results to the Security Tab and not be raised by people's PRs or landings.  This is something that needs to be taken up with @ryon-jensen and/or the team in general to define the process and responsible part for monitoring the Security Tab.

Reply:
> > I have observed some limits on the Trivy database access. ([daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21](https://github.com/daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21))
> 
> 44000/minute seems like a lot. Are we sure something did not run amok?

I guess this is a limit for all Github runners - let's check in the future how many false-positive we will have.
> 
> > Everything that breakes the PR build, but is not related to PR-related changes is VERY expensive from the PR's author perspective.
> 
> No more than the status quo. Just look at the **isort** action/check that has been broken for a few days last week. It finally (and only, I suspect!) got updated because somebody noticed it in their PR and started asking about it/getting it fixed. If it did not break a PR I am pretty sure it would still be broken and nobody noticing.
> 
> > Moreover, the same problem may hit several PRs in parallel. Who will be responsible to fix it?
> 
> Just like the **isort** problem last week. Somebody will raise their PR being broken because of it and somebody jumps in to fix it.
> 
"Somebody will raise it ... somebody jumps in to fix it"  is not a well-defined process :(. But let's keep it as it is for a while and see how it works.

> > Trivy is configured intentionally to detect CVEs in Python/Go/Java. This defines scope of files to be scaned. I do not expect that this will change soon.
> 
> Which is even worse. Because when a long time goes by and they do finally change something, nobody remembers that we need to change on our end also to capture that new something.
> 
> > Why we can not treat `schedule` in the same way as we treat daily tests? I expect that one scan per day is enough in normal case.
> 
> Phil has been assigned ownership of monitoring daily scans. Who will be assigned daily monitoring of the Security Tab? This is something that needs to be addressed before we introduce a security scan that may **only** publish results to the Security Tab and not be raised by people's PRs or landings. This is something that needs to be taken up with @ryon-jensen and/or the team in general to define the process and responsible part for monitoring the Security Tab.

Let's keep the `schedule` trigger just in case the `push` (to master) fails and we do not have any results updated in the Security tab until next PR lands.

Reply:
> 44000/minute seems like a lot. Are we sure something did not run amok?

Trivy database overuse is a very common issue. The problem has been resolved by introducing a caching mechanism in:
https://github.com/daos-stack/daos/pull/15201/commits/184fabd8dd19628de116749927387b66e1d6cbe7

Reply:
> Please take a look at [grom72/daos/actions/caches](https://github.com/grom72/daos/actions/caches/)

Surely this cache space is not unlimited, yes?  How do old cache entries get evicted?

So we are only refreshing this cache every 24h.  Doesn't that open up a window where a PR does not fail on something that could be landed the next day and fail when landed because the cache was stale?  I suppose this is always a problem even without caching though.

> and [grom72/daos/actions/runs/11549046811](https://github.com/grom72/daos/actions/runs/11549046811) to see how the solution works.

Unfortunately that one just exits because it's a PR and therefore not on the default branch.  More comments in the review.